[X86] Fix typo: QWORD alignment is greater than or equal to 8, not greater than 8

AZero13 opened the pull request:
Align(8) is QWORD aligned, but this was checking to see if alignment was greater than that, when it should have been checking for being greater than OR EQUAL to Align(8).

This bug was introduced in https://github.com/llvm/llvm-project/commit/6a6af30d433d7 during the transition to the Align type.

topperc commented:
But no tests are affected?

AtariDreams commented:
> But no tests are affected?

Align(8) is QWORD aligned, but this was checking to see f alignment was higher than that, which it is not.

Another PR I am working on for optsize depends on this, but because it is an oversight or typo that affects the compiler, I figure I would make this into a separate PR.

topperc commented:
> > But no tests are affected?
> 
> Align(8) is QWORD aligned, but this was checking to see f alignment was higher than that, which it is not.
> 
> Another PR I am working on for optsize depends on this, but because it is an oversight or typo that affects the compiler, I figure I would make this into a separate PR.

Your description said "This probably inhibited a lot of optimizations.". That should be pretty easy to prove so why no test updates?

AZero13 committed:
[X86] Pre-commit tests (NFC)

AtariDreams commented:
> > > But no tests are affected?
> 
> > 
> 
> > Align(8) is QWORD aligned, but this was checking to see f alignment was higher than that, which it is not.
> 
> > 
> 
> > Another PR I am working on for optsize depends on this, but because it is an oversight or typo that affects the compiler, I figure I would make this into a separate PR.
> 
> 
> 
> Your original description said "This probably inhibited a lot of optimizations.". That should be pretty easy to prove so why no test updates?

Done!

topperc commented on llvm/lib/Target/X86/X86SelectionDAGInfo.cpp:
```diff
@@ -66,8 +66,10 @@ SDValue X86SelectionDAGInfo::EmitTargetCodeForMemset(
   // address value and run time information about the CPU.
-  if (Alignment < Align(4) || !ConstantSize ||
-      ConstantSize->getZExtValue() > Subtarget.getMaxInlineSizeThreshold()) 
+  if (!ConstantSize ||
```
This change does not match the description of the patch.

AtariDreams replied:
I'm gonna then branch that to another PR because it's needed

https://github.com/llvm/llvm-project/pull/87825

AZero13 committed:
[X86] Fix typo: QWORD alignment is greater than or equal to 8, not greater than 8

Align(8) is QWORD aligned, but this was checking to see if alignment was greater than that, when it should have been checking for being greater than OR EQUAL to Align(8).

This bug was introduced in commit 6a6af30d433d7 during the transition to the Align type.

RKSimon commented on llvm/test/CodeGen/X86/memset-minsize.ll:
```diff
@@ -136,4 +136,18 @@ entry:
+}
+
 declare void @llvm.memset.p0.i32(ptr nocapture writeonly, i8, i32, i1)
+declare void @llvm.memset.inline.p0.i64(ptr nocapture writeonly, i8, i64, i1)
```
remove this or fix this? we don't need to declare anymore (hence why it still worked), but even if you want to keep it, you should only declare used intrinsics

phoebewang commented on llvm/lib/Target/X86/X86SelectionDAGInfo.cpp:
```diff
@@ -86,7 +86,7 @@ SDValue X86SelectionDAGInfo::EmitTargetCodeForMemset(
       Val = (Val << 8)  | Val;
       Val = (Val << 16) | Val;
-      if (Subtarget.is64Bit() && Alignment > Align(8)) { // QWORD aligned
+      if (Subtarget.is64Bit() && Alignment > Align(4)) { // QWORD aligned
```
Maybe use `>= Align(8)` for better understanding?

AtariDreams replied:
I'm trying to mimic the style of the function I'm modifying (which does > Align(2) but I guess that also works.

AtariDreams commented:
@phoebewang ping?

AtariDreams commented:
I don't have commit access by the way.

AtariDreams commented:
/cherry-pick https://github.com/llvm/llvm-project/commit/8389b3bf60ef3fbd04c6efc5ff4d4605d10e7fc5
